scipy-stubs/sparse/_construct.pyi

```diff
@@ -76,7 +77,15 @@ _Offsets: TypeAlias = onp.ToInt | onp.ToInt1D
 
 _DataRVS: TypeAlias = Callable[[int], onp.ArrayND[_Numeric]]
 
-_ToBlocks: TypeAlias = Seq[Seq[_spbase[_SCT] | None]] | onp.ArrayND[np.object_]
+_ToBlocksArray: TypeAlias = onp.ArrayND[np.object_]
+_ToBlocksSPArray: TypeAlias = Seq[Seq[_SpArray2D[_SCT] | None]]
+_ToBlocksSPMatrix: TypeAlias = Seq[Seq[_SpMatrix[_SCT] | None]]
+_ToBlocksCanStack: TypeAlias = Seq[Seq[_CanStack[_T] | None]]
+_ToBlocksCanStackAs: TypeAlias = Seq[Seq[_CanStackAs[_SCT0, _T] | None]]
+_ToBlocksUnkown: TypeAlias = _ToBlocksSPArray | _ToBlocksArray
+
+_ToMatsDiagKnown: TypeAlias = Iterable[_spbase[_SCT] | onp.ToArrayND[_SCT]]
```

Comment:
I suppose that because the generic type parameter already implies that the type is known, there's no real need for that explicit `*Known` suffix. But it's also fine to leave it like this AFAIK 🤷🏻